vispy/app/backends/_tk.py

```diff
@@ -0,0 +1,498 @@
+            pos=(e.x, e.y), modifiers=self._parse_state(e))
+        
+    def _on_mouse_button_press(self, e):
+        if self._vispy_canvas is None:
+            return
+        # [ left=1, middle, right]
+        btn = { 1:1, 2:3, 3:2}
+        self._vispy_mouse_press(
+            pos=(e.x, e.y), button=btn[e.num], modifiers=self._parse_state(e))
+        
+    def _on_mouse_double_button_press(self, e):
+        if self._vispy_canvas is None:
+            return
+        # [ left=1, middle, right]
+        btn = { 1:1, 2:3, 3:2}
+        pos, button, modifiers = \
+            (e.x, e.y), btn[e.num], self._parse_state(e)
+        self._vispy_mouse_press(
+            pos=pos, button=button, modifiers=modifiers)
+        self._vispy_mouse_double_click(
+            pos=pos, button=button, modifiers=modifiers)
+
+    def _on_mouse_button_release(self, e):
+        if self._vispy_canvas is None:
+            return
+        # [ left=1, middle, right]
+        btn = { 1:1, 2:3, 3:2}
+        self._vispy_mouse_release(
+            pos=(e.x, e.y), button=btn[e.num], modifiers=self._parse_state(e))
+    
+    def _on_configure(self, e):
+        print("_on_configure")
+        if self._vispy_canvas is None or not self._init:
+            return
+        self.update()
+        self.tkResize(e)
+        self._vispy_canvas.events.resize(size=(e.width, e.height))
+    
+    def _on_draw(self, e):
+        if self._vispy_canvas is None:
+            return
+        if not self._init:
+            self._initialize()
+        self._vispy_canvas.set_current()
+        self._vispy_canvas.events.draw(region=None)
+    
+    def _initialize(self):
+        print("_initialize")
+        self.initgl()
+        
+        if self._vispy_canvas is None:
+            return
+        self._init = True
+        self._vispy_canvas.set_current()
+        self._vispy_canvas.events.initialize()
+        if self.top: self.top.update_idletasks()
+        self._on_configure(Coord(self._vispy_get_size()))
+    
+    def _on_key_down(self, e):
+        if self._vispy_canvas is None:
+            return
+        key, text = self._parse_keys(e)
+        self._vispy_canvas.events.key_press(
+            key=key, text=text, modifiers=self._parse_state(e))
+        
+    def _on_key_up(self, e):
+        if self._vispy_canvas is None:
+            return
+        key, text = self._parse_keys(e)
+        self._vispy_canvas.events.key_release(
+            key=key, text=text, modifiers=self._parse_state(e))
+
+    def _vispy_warmup(self):
+        etime = time() + 0.3
+        while time() < etime:
+            sleep(0.01)
+            self._vispy_canvas.set_current()
+            self._vispy_canvas.app.process_events()
+
+    def _vispy_set_current(self):
+        # Make this the current context
+        self.tkMakeCurrent()
+
+    def _vispy_swap_buffers(self):
+        # Swap front and back buffer
+        self._vispy_canvas.set_current()
+        self.tkSwapBuffers()
```

Comment:
What happens if you comment out this line? I don't see a swap buffers call in the pyopengltk demo.py file so maybe it is auto-swapping?

Reply:
Same thing, screen stays black.